backend/apps/core/tasks.py

```
@@ -218,3 +221,101 @@ def _purge_for_model(model, threshold_date, batch_size):
+        env["PGPASSWORD"] = db_settings["PASSWORD"]
+
+    cmd = [
+        "pg_dump",
+        "--host", db_settings.get("HOST", "localhost"),
+        "--port", str(db_settings.get("PORT") or 5432),
+        "--username", db_settings.get("USER", ""),
+        "--dbname", db_settings.get("NAME", ""),
+        "--no-password",
+        "--file", str(output_path),
+    ]
+    subprocess.run(cmd, env=env, check=True, capture_output=True)
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)